Tests/Packet++Test/Tests/DnsTests.cpp

```diff
@@ -338,11 +338,12 @@ PTF_TEST_CASE(DnsLayerResourceCreationTest)
 	PTF_ASSERT_EQUAL(dnsEdit4Packet.getLayerOfType<pcpp::DnsLayer>()->getFirstAnswer(), firstAnswer, ptr);
 	PTF_ASSERT_EQUAL(dnsEdit4Packet.getLayerOfType<pcpp::DnsLayer>()->getNextAnswer(firstAnswer), secondAnswer, ptr);
 
+	PTF_ASSERT_RAISE_INCLUDES(pcpp::IPv4DnsResourceData(std::string("256.249.90.238")), std::invalid_argument, "Not a valid IPv4 address");
```

Comment:
Why do we want to use `PTF_ASSERT_RAISE_INCLUDES` and not `PTF_ASSERT_RAISE`? We know the expected message with the IP address. Same in the test of the tests

Reply:
I'm not sure I understand. In this test and most (if not all) of the others we know exactly what the error message would be because we control the input (in this case: `pcpp::IPv4DnsResourceData(std::string("256.249.90.238"))`), why not asserting the entire error message? 🤔

Reply:
@seladb  you are correct, we do know the whole error message. However, here is my consideration:

Say if we have `IPv4Network(address, "invalid_netmask")`: `IPv4Address("invalid_netmask")` will throw the exception, `"Not a valid IPv4 address: invalid_netmask"`, and then `IPv4Network` captures it and throw again with the message, `"netmask is not valid: Not a valid IPv4 address: invalid_netmask"`

Is it necessary to capture the entire error message for `PTF_ASSERT_RAISE` here? Perhaps we only need to focus on the "netmask is not valid" error. The key consideration is whether documenting the complete error message is necessary, especially when it arises from a chain of exceptions, which might seem trivial.

Reply:
Please see this comment: https://github.com/seladb/PcapPlusPlus/pull/1345#discussion_r1546035739 . Getting the full message actually helps understand if the user gets an exception message that makes sense or not. When it's not, we should consider changing the code to make it more readable or less repetative

Reply:
@seladb  sure, let me make the error message clear an simple.